commons/protocol/src/main/scala/sbt/protocol/Keys.scala

```diff
@@ -82,11 +82,29 @@ object KeyList {
   implicit val pickler: Pickler[KeyList] = genPickler[KeyList]
 }
 
+// TODO - This is a bad name...
+/** The notion of classpath dependencies between projects. */
+final case class ClasspathDep(
+  project: ProjectReference,
+  configuration: Option[String])
+object ClasspathDep {
+  implicit val picklerUnpickler = PicklerUnpickler.generate[ClasspathDep]
+}
+/** The notiion of build dependencies between projects. */
+final case class ProjectDependencies(
+  classpath: Seq[ClasspathDep],
+  aggregate: Seq[ProjectReference])
+object ProjectDependencies {
+  implicit val picklerUnpickler = PicklerUnpickler.generate[ProjectDependencies]
+}
+
 /** Core information returned about projects for build clients. */
 final case class MinimalProjectStructure(
   id: ProjectReference,
   // Class names of plugins used by this project.
-  plugins: Vector[String])
+  plugins: Vector[String],
+  /** A possible list of dependencies between projects..   Added in 0.2 of the protocol. */
```

Comment:
protocol version right now isn't necessarily tied to the jar version, fwiw. I don't think it's clear how we want to use the protocol version. at the moment the protocol version is simply an integer, not a semantic version with multiple components. breaking changes in the protocol aren't allowed at all ... conceptually the protocol version is probably the "Y" in the semver but I'm not sure.